app/src/main/java/ca/marshallasch/veil/DataStore.java

```diff
@@ -0,0 +1,198 @@
+     * @param post the post to insert
+     * @return true on success false oon failure.
+     */
+    public boolean savePost(@Nullable DhtProto.Post post) {
+
+        if (post == null) {
+            return false;
+        }
+
+        String postHash = hashTableStore.insertPost(post);
+
+        // insert any necessary query keys in here.
+
+        return db.insertKnownPost(postHash, null);
+    }
+
+    /**
+     * Gets all the known posts in a the data store
+     * @return the list of posts pairs.
+     */
+    public List<Pair<String, DhtProto.Post>> getKnownPosts() {
+
+        List<String> hashes = db.getPostHashes();
+
+        Pair<String, DhtProto.Post> postPair;
+        List<Pair<String, DhtProto.Post>> posts = new ArrayList<>();
+
+        // get each post that is in the list
+        for (String hash: hashes) {
+
+            postPair = null;
+            try {
+                postPair = hashTableStore.findPostByHash(hash);
+            }
+            catch (TooManyResultsException e) {
+                e.printStackTrace();
+            }
+
+            // add the post to the list
+            if (postPair != null) {
+                posts.add(postPair);
+            }
+        }
+
+        return posts;
+    }
+
+    /**
+     * Gets the list of comments for a post sorted in by time posted for a given post.
+     * @param postHash the hash of the post that the comments will be found for.
+     * @return a list of comment pairs
+     */
+    public List<Pair<String, DhtProto.Comment>> getCommentsForPost(String postHash) {
+
+        List<String> hashes = db.getCommentHashes(postHash);
+
+        Pair<String, DhtProto.Comment> commentPair;
+        List<Pair<String, DhtProto.Comment>> comments = new ArrayList<>();
+
+        // get each post that is in the list
+        for (String hash: hashes) {
+
+            commentPair = null;
+            try {
+                commentPair = hashTableStore.findCommentByHash(hash);
+            }
+            catch (TooManyResultsException e) {
+                e.printStackTrace();
+            }
+
+            // add the post to the list
+            if (commentPair != null) {
+                comments.add(commentPair);
+            }
+        }
+
+        Collections.sort(comments, new CommentPairComparator());
+
+        return comments;
+
+    }
+
+
+    public Sync.MappingMessage getDatabase() {
+
+        List<Pair<String, String>> knownPosts = db.dumpKnownPosts();
+
+        Sync.MappingMessage.Builder builder = Sync.MappingMessage.newBuilder();
+
+        // add it to the list
+        for (Pair<String, String> pair: knownPosts) {
+            builder.addMappings(Sync.CommentMapping.newBuilder()
+                    .setPostHash(pair.first)
+                    .setCommentHash(pair.second)
+                    .build());
+        }
+
+        // build the message to send to other devices
+        return builder.build();
+    }
+
+    public Sync.HashData getDataStore() {
```

Comment:
Add javadoc comment about the functions of this method.

Reply:
Fixed in commit 6d83384